### PR TITLE
FVLC-45

### PR DESCRIPTION
**Pull request by jonathanwbayer**
hooray

### CODE REVIEW

**Comment by adam-vessey** [comment]
Coding standards:

```
vagrant@whip-vagrant-i7latest:/var/www/drupal7/sites/all/modules/islandora_solution_pack_serial$ drush-test-no-empty islandora_serial_object
Release standards, Drupal Security Checks, Drupal Coding Standards, Internationalization, Interface text translatability, Drupal CodeSniffer

sites/all/modules/islandora_solution_pack_serial/islandora_serial_object.module:
 +49: [minor] Inline comments must end in full-stops, exclamation marks, or question marks
 +56: [minor] Line indented incorrectly; expected 2 spaces, found 0
 +88: [normal] Use an indent of 2 spaces, with no tabs
 +88: [minor] Line indented incorrectly; expected 2 spaces, found 1
 +88: [minor] Spaces must be used to indent lines; tabs are not allowed
 +304: [normal] Use an indent of 2 spaces, with no tabs
 +304: [minor] Line indented incorrectly; expected 4 spaces, found 1
 +304: [minor] Spaces must be used to indent lines; tabs are not allowed
 +305: [normal] Use an indent of 2 spaces, with no tabs
 +305: [minor] Line indented incorrectly; expected 4 spaces, found 1
 +305: [minor] Spaces must be used to indent lines; tabs are not allowed
 +306: [normal] Use an indent of 2 spaces, with no tabs
 +306: [minor] Line indented incorrectly; expected 4 spaces, found 1
 +306: [minor] Spaces must be used to indent lines; tabs are not allowed
 +307: [normal] Use an indent of 2 spaces, with no tabs
 +307: [minor] Array indentation error, expected 8 spaces but found 7
 +307: [minor] Spaces must be used to indent lines; tabs are not allowed
 +308: [normal] Use an indent of 2 spaces, with no tabs
 +308: [minor] Line indented incorrectly; expected 4 spaces, found 1
 +308: [minor] Spaces must be used to indent lines; tabs are not allowed

sites/all/modules/islandora_solution_pack_serial/includes/pdf.form.inc:
 +126: [minor] There should be no trailing spaces
 +126: [minor] Whitespace found at end of line
 +220: [normal] use a space between the closing parenthesis and the open bracket
 +220: [minor] Missing function doc comment
 +220: [minor] Expected 1 space after closing parenthesis; found 0
 +225: [minor] There should be no trailing spaces
 +225: [normal] Use uppercase for PHP constants, e.g. NULL, TRUE, FALSE
 +225: [minor] TRUE, FALSE and NULL must be uppercase; expected "TRUE" but found "true"
 +225: [minor] Whitespace found at end of line
 +226: [minor] A comma should follow the last multiline array item. Found: )
 +231: [normal] use a space between the closing parenthesis and the open bracket
 +231: [minor] Missing function doc comment
 +231: [minor] Expected 1 space after closing parenthesis; found 0
 +232: [minor] There should be no trailing spaces
 +232: [minor] Whitespace found at end of line
 +237: [minor] Files must end in a single new line character

Status Messages:
 Coder found 1 projects, 12 files, 9 normal warnings, 27 minor warnings, 0 warnings were flagged to be ignored

Coder found 1 projects, 12 files, 9 normal warnings, 27 minor warnings, 0 warnings were flagged to  [status]
be ignored
Drush command terminated abnormally due to an unrecoverable error.                                  [error]
vagrant@whip-vagrant-i7latest:/var/www/drupal7/sites/all/modules/islandora_solution_pack_serial$
```

File: includes/pdf.form.inc [hunk 1]
```diff
@@ -197,3 +216,21 @@ function islandora_serial_object_set_pdf_position($pid, $position) {
...
+      '#required' => true, 
+    )
+  );
+  return $form;
+}
+
+function islandora_serial_object_ingest_section_form_submit($form, &$form_state){
+  $obj = islandora_ingest_form_get_object($form_state);  
+  $obj->label = $form_state['values']['title'];
+  return $form;
```
Comment (adam-vessey) [inline includes/pdf.form.inc]: Submission handlers do not have a return.

Comment (adam-vessey) [inline includes/pdf.form.inc]: (Or at least, the return value is not used)

**Comment by adam-vessey** [comment]
Perhaps I'm not 100% understanding this... But, are you supposed to be able to add PDFs to the "intermediate stubs"?  If so, [the access callback](https://github.com/discoverygarden/islandora_solution_pack_serial/pull/19/files#diff-6265f6c799f9f356d85184a472de6964L215) likely has to change to include the new content model.

**Comment by jonathanwbayer** [comment]
you aren't supposed to be able toa dd PDF's to the intermediate stubs. they are to function as "sections" for the heirarchal PDF displays.

File: includes/pdf.form.inc [hunk 2]
```diff
@@ -197,3 +215,31 @@ function islandora_serial_object_set_pdf_position($pid, $position) {
...
+ * Ingest step callback to name an object .
+ *
+ * @param array $form_state
+ *   An array containing the Drupal form state.
+ */
+function islandora_serial_object_ingest_section_form_submit($form, &$form_state) {
+  $obj = islandora_ingest_form_get_object($form_state);
+  $obj->label = $form_state['values']['title'];
+}
+
```
Comment (adam-vessey) [inline includes/pdf.form.inc]: An extra new line here?:

```
vagrant@whip-vagrant-i7latest:/var/www/drupal7/sites/all/modules/islandora_solution_pack_serial$ drush-test-no-empty islandora_serial_object
Release standards, Drupal Security Checks, Drupal Coding Standards, Internationalization, Interface text translatability, Drupal CodeSniffer

sites/all/modules/islandora_solution_pack_serial/includes/pdf.form.inc:
 +246: [minor] Files must end in a single new line character

Status Messages:
 Coder found 1 projects, 12 files, 1 minor warnings, 0 warnings were flagged to be ignored

Coder found 1 projects, 12 files, 1 minor warnings, 0 warnings were flagged to be ignored                                                                                        [status]
Drush command terminated abnormally due to an unrecoverable error.
```

File: islandora_serial_object.module [hunk 3]
```diff
@@ -78,6 +85,8 @@ function islandora_serial_object_islandora_required_objects(IslandoraTuque $conn
       'objects' => array(
         $serial_content_model,
         $intermediate_content_model,
+        $intermediate_content_model_s,
+
```
Comment (adam-vessey) [inline islandora_serial_object.module]: Empty line seems weird.

File: includes/pdf.form.inc (hunk 2 above)
Comment (adam-vessey) [inline includes/pdf.form.inc]: As in, it should be removed... _mumble_

**Pull request merged**